src/App.tsx

```diff
@@ -1,61 +1,137 @@
+import { Todo } from './interfaces/Todo';
+
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { useState } from 'react';
+import { TodoList } from './components/TodoList';
+
+const initialTodos: Todo[] = todosFromServer.map(todo => {
+  const user = usersFromServer.find(client => client.id === todo.userId)!;
+
+  return {
+    ...todo,
+    user,
+  };
+});
 
 export const App = () => {
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>(initialTodos);
+
+  const [selectedUser, setSelectedUser] = useState<number>(0);
+  const [selectedUserError, setSelectedUserError] = useState(false);
+
+  const [title, setTitle] = useState<string>('');
+  const [titleError, setTitleError] = useState(false);
+
+  const addTodo = () => {
+    setVisibleTodos(current => {
+      const id = Math.max(...current.map(item => item.id));
+      const user = usersFromServer.find(person => person.id === selectedUser);
```

Comment:
[CODE STYLE] - Don't repeat yourself. You already found the user for the todos in the initial mapping (lines 10-17). Consider storing that information so you don't have to find the user again when adding a new todo. This will help in reducing redundancy and improving performance.